tests/unit/utils/test_module_indexer.py

```diff
@@ -0,0 +1,278 @@
+            'lib/ansible/modules/bar/foofoo.py': None,
+            'lib/ansible/modules/baz/test/code.py': None,
+        }
+
+        expected = {}
+        key = 'lib/ansible/modules/foo/bar'
+        expected[key] = {
+            'maintainers': ['jim'],
+            'maintainers_keys': [key],
+        }
+        key = 'lib/ansible/modules/foo/bar/baz.py'
+        expected[key] = {
+            'maintainers': ['bob', 'jim'],
+            'maintainers_keys': ['lib/ansible/modules/foo/bar', key],
+        }
+        key = 'lib/ansible/modules/foo2/bar2'
+        expected[key] = {
+            'maintainers': ['jim'],
+            'maintainers_keys': [key],
+        }
+        key = 'lib/ansible/modules/foo2/bar2/baz.py'
+        expected[key] = {
+            'maintainers': ['bob', 'jim'],
+            'maintainers_keys': ['lib/ansible/modules/foo2/bar2', key],
+        }
+        key = 'lib/ansible/modules/bar/foo'
+        expected[key] = {
+            'maintainers': ['csim'],
+            'maintainers_keys': [key],
+        }
+
+        key = 'lib/ansible/modules/bar/foofoo.py'
+        expected[key] = {
+            'maintainers': ['uolip'],
+            'maintainers_keys': [key],
+        }
+
+        key = 'lib/ansible/modules/baz'
+        expected[key] = {
+            'maintainers': ['Loulou', 'ZaZa'],
+            'maintainers_keys': [key],
+        }
+
+        key = 'lib/ansible/modules/baz/test'
+        expected[key] = {
+            'maintainers': ['Loulou', 'jim', 'ZaZa'],
+            'maintainers_keys': ['lib/ansible/modules/baz', key],
+        }
+
+        key = 'lib/ansible/modules/baz/test/code.py'
+        expected[key] = {
+            'maintainers': ['Loulou', 'bob', 'jim', 'ZaZa'],
+            'maintainers_keys': ['lib/ansible/modules/baz', 'lib/ansible/modules/baz/test', key],
+        }
+
+        indexer = run(textwrap.dedent(self.BOTMETA), filepaths)
+
+        self.assertEqual(len(indexer.modules), len(filepaths))  # ensure only fake data are loaded
+
+        for k in expected:
+            # BOTMETA and modules have identical maintainers since there is no authors
+            # defined in source code
+            self.assertEqual(sorted(indexer.botmeta['files'][k]['maintainers_keys']),sorted(expected[k]['maintainers_keys']))
+            self.assertEqual(sorted(indexer.botmeta['files'][k]['maintainers']), sorted(expected[k]['maintainers']))
+
+        for k in filepaths:
+            self.assertEqual(sorted(indexer.modules[k]['maintainers_keys']),sorted(expected[k]['maintainers_keys']))
+            self.assertEqual(sorted(indexer.modules[k]['maintainers']), sorted(expected[k]['maintainers']))
+
+    def test_authors_overridden_if_entry_in_BOTMETA(self):
+        """Check that authors are overidden if BOTMETA contains an entry.
+
+        Ensure that authors defined in 'author' field of 'DOCUMENTATION' module
+        metadata are omitted when there is a matching entry in BOTMETA.yml.
```

Comment:
authors should be included, not overridden. If an author does not wish to be a maintainer, they should be added to the "ignored' field in botmeta OR be removed as an author.

Reply:
Pull request updated. Note that currently, in this case, authors are overridden.

I much prefer the behavior you expect.